### ISSUE TITLE
出现了bbox被0除的问题

### ISSUE THREAD

**Issue by CindyHuang23** [DetectionTeamUCAS/RetinaNet_Tensorflow_Rotation#9]
/home/lyy/hq/RetinaNet_Tensorflow_Rotation-master/libs/box_utils/bbox_transform.py:99: RuntimeWarning: divide by zero encountered in log
  targets_dh = np.log(gt_rois[:, 3] / ex_rois[:, 3])
训练过程中出现了这种错误，请问怎么解决呢？

**Comment by yangxue0827** [comment]
贴一下cfgs.py @CindyHuang23

**Comment by CindyHuang23** [comment]
# ------------------------------------------------
VERSION = 'RetinaNet_v1'#================================================modify
NET_NAME = 'resnet_v1_101'  # 'MobilenetV2'
ADD_BOX_IN_TENSORBOARD = True

# ---------------------------------------- System_config
ROOT_PATH = os.path.abspath('../')
print(20*"++--")
print(ROOT_PATH)
GPU_GROUP = "0"
NUM_GPU = len(GPU_GROUP.strip().split(','))
SHOW_TRAIN_INFO_INTE = 20
SMRY_ITER = 200
SAVE_WEIGHTS_INTE = 27000

SUMMARY_PATH = ROOT_PATH + '/output/summary'
TEST_SAVE_PATH = ROOT_PATH + '/tools/test_result'

if NET_NAME.startswith("resnet"):
    weights_name = NET_NAME
elif NET_NAME.startswith("MobilenetV2"):
    weights_name = "mobilenet/mobilenet_v2_1.0_224"
else:
    raise Exception('net name must in [resnet_v1_101, resnet_v1_50, MobilenetV2]')

PRETRAINED_CKPT = ROOT_PATH + '/data/pretrained_weights/' + weights_name + '.ckpt'
TRAINED_CKPT = os.path.join(ROOT_PATH, 'output/trained_weights')
EVALUATE_DIR = ROOT_PATH + '/output/evaluate_result_pickle/'

# ------------------------------------------ Train config
RESTORE_FROM_RPN = False
FIXED_BLOCKS = 1  # allow 0~3
FREEZE_BLOCKS = [True, False, False, False, False]  # for gluoncv backbone
USE_07_METRIC = True

MUTILPY_BIAS_GRADIENT = 2.0  # if None, will not multipy
GRADIENT_CLIPPING_BY_NORM = 10.0  # if None, will not clip

CLS_WEIGHT = 1.0
REG_WEIGHT = 1.0
USE_IOU_FACTOR = True

BATCH_SIZE = 1
EPSILON = 1e-5
MOMENTUM = 0.9
LR = 5e-4  # * NUM_GPU * BATCH_SIZE
DECAY_STEP = [SAVE_WEIGHTS_INTE*12, SAVE_WEIGHTS_INTE*16, SAVE_WEIGHTS_INTE*20]
MAX_ITERATION = SAVE_WEIGHTS_INTE*20
WARM_SETP = int(1.0 / 8.0 * SAVE_WEIGHTS_INTE)

# -------------------------------------------- Data_preprocess_config
DATASET_NAME = 'RSS_DATA'  # 'pascal', 'coco'#==================================================================modify
PIXEL_MEAN = [123.68, 116.779, 103.939]  # R, G, B. In tf, channel is RGB. In openCV, channel is BGR
PIXEL_MEAN_ = [0.485, 0.456, 0.406]
PIXEL_STD = [0.229, 0.224, 0.225]  # R, G, B. In tf, channel is RGB. In openCV, channel is BGR
IMG_SHORT_SIDE_LEN = 512#=========================================================================================modify
IMG_MAX_LENGTH = 512#=========================================================================================================modify
CLASS_NUM = 18#==========================================================================================modify

# --------------------------------------------- Network_config
SUBNETS_WEIGHTS_INITIALIZER = tf.random_normal_initializer(mean=0.0, stddev=0.01, seed=None)
SUBNETS_BIAS_INITIALIZER = tf.constant_initializer(value=0.0)
PROBABILITY = 0.01
FINAL_CONV_BIAS_INITIALIZER = tf.constant_initializer(value=-math.log((1.0 - PROBABILITY) / PROBABILITY))
WEIGHT_DECAY = 1e-4

# ---------------------------------------------Anchor config
LEVEL = ['P3', 'P4', 'P5', 'P6', 'P7']
BASE_ANCHOR_SIZE_LIST = [32, 64, 128, 256, 512]
ANCHOR_STRIDE = [8, 16, 32, 64, 128]
ANCHOR_SCALES = [2 ** 0, 2 ** (1.0 / 3.0), 2 ** (2.0 / 3.0)]
ANCHOR_RATIOS = [1, 1 / 3., 3., 5., 1 / 5.]
ANCHOR_ANGLES = [-90, -75, -60, -45, -30, -15]
ANCHOR_SCALE_FACTORS = None
USE_CENTER_OFFSET = True
METHOD = 'R'
USE_ANGLE_COND = False

# --------------------------------------------RPN config
SHARE_NET = True
USE_P5 = True
IOU_POSITIVE_THRESHOLD = 0.5
IOU_NEGATIVE_THRESHOLD = 0.4

NMS = True
NMS_IOU_THRESHOLD = 0.5
MAXIMUM_DETECTIONS = 100
FILTERED_SCORE = 0.05
VIS_SCORE = 0.4

# --------------------------------------------NAS FPN config
NUM_FPN = 0
NUM_NAS_FPN = 0
USE_RELU = True
FPN_CHANNEL = 256

**Comment by CindyHuang23** [comment]
************************************************************
2019-07-26 13:24:33: global_step:2180  current_step:2180
per_cost_time:0.410s
cls_loss:1.283985
reg_loss:nan
total_losses:nan
训练中还出现了这个问题。。。不知道是怎么回事

**Comment by yangxue0827** [comment]
你有训练过我给的默认配置文件吗，我不清楚是否是本身代码的问题还是数据集不同带来的问题 @CindyHuang23

**Comment by CindyHuang23** [comment]
没有，但是我是按照你的read me 文件修改了部分参数，修改的地方配置文件有注释

**Comment by yangxue0827** [comment]
我会更新一下代码 @CindyHuang23

**Comment by yangxue0827** [comment]
代码已更新，建议先跑通一个默认的配置文件 @CindyHuang23

**Comment by zhouzhubin** [comment]
@CindyHuang23 你好，请问你可以了吗？我把模型尺寸改成400了，训练到900step就必出现了，我也是用的自己的数据集。我另外训练了一个600的，就出RuntimeWarning: divide by zero encountered in log targets_dh = np.log(gt_rois[:, 3] / ex_rois[:, 3])一次，但是loss是好的，没有出现nan
